modules/ROOT/pages/map-users-roles-ldap-task.adoc

```diff
@@ -1,39 +1,152 @@
+
+For example, your IDP may provide your groups in the following format:
+
+----
+<ns2:Attribute Name="Groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
+  <ns2:AttributeValue>Mule_Org_Admin_XXX</ns2:AttributeValue>
+</ns2:Attribute>
+<ns2:Attribute Name="Groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
+  <ns2:AttributeValue>Mule_Exchange_XXX</ns2:AttributeValue>
+</ns2:Attribute>
+----
+
+To prevent security vulnerabilities, Anypoint Platform requires you to create a single group that encompasses multiple group mappings:
+
+----
+<ns2:Attribute Name="Groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
+  <ns2:AttributeValue>Mule_Org_Admin_XXX</ns2:AttributeValue>
+  <ns2:AttributeValue>Mule_Exchange_XXX</ns2:AttributeValue>
+</ns2:Attribute>
+----
+. In Anypoint Platform, navigate to Access Management.
+. Click *Identity Providers*.
+. From *Identity Providers*, select OIDC.
+. Verify that the value in the *Group Attribute* field matches the name of the SAML attribute that contains the group names
+. In the above example of a SAML assertion, the *Group Attribute* is "Groups":
+=== Map SAML Users to Roles
 
 . In Anypoint Platform, navigate to Access Management.
 . Click *Roles*.
 . Click *Add Role* to create a role for each group of users in your organization.
-+
-image::external-identity-34af9.png[]
-
 . Specify a role name and description.
-. Click *Add Role* to add the role:
-+
-image::external-identity-c731b.png[]
+. Click *Add Role* to add the role.
+. In the *Roles* section, click the name of the new role:
+. Click *Set external group mapping*.
+. Enter the group to which you want to map.
+. Click *Set names*.
 +
+To map more than one attribute name to the selected role, click *Add More* and add another attribute.
+
+. Repeat this process for each role that you want to map to an external group.
+
+== Configure Multiple Group Mappings for Single Sign-On using OpenID Connect (OIDC)
+
+External identity providers format groups claims in different ways.
+Anypoint Platform checks the `UserInfo` response for the groups claim. If it is not found, it then checks the `Token` response.
 
+You can use any identity providers that return groups claims on the `UserInfo` or `Token` response.
+
+=== Map OIDC Group Users to Roles
+
+. Obtain a response to the `UserInfo` or `Token` endpoint that contains a groups claim from your external identity provider.
+. In Anypoint Platform, navigate to Access Management.
+. Click *Identity Providers*.
+. From *Identity Providers*, select OIDC.
+. If your IdP requires an additional OIDC scope to authorize access to groups claims, enter it in the *Group Scope* field.
+. Verify that the JSONata query in the *Group Attribute JSONata Expression* yields an array of group names from either the `UserInfo` or `Token` response.
+. Click *Roles*.
+. Click *Add Role* to create a role for each group of users in your organization.
```

Comment:
Clarify: Is this step and the remainder of this task addressing one REPEATED set of steps to create a role for EACH group, or one set of steps that creates a role of ALL of the groups?